src/tools/web/kdb.js

```diff
@@ -0,0 +1,116 @@
+// kdb.js - small library to access Elektra's kdb via node.js
+const { exec } = require('child_process')
+
+// constants
+const ERR_KEY_NOT_FOUND = 'Did not find key'
+
+// KDBError
+function KDBError (message) {
+    this.name = 'KDBError'
+    this.message = message || ''
+}
+KDBError.prototype = Error.prototype
+
+// remove newline from the end of a string
+const trimNewline = (str) =>
+  str.substring(0, str.length - 1)
+
+// execute a script while catching and parsing errors
+const safeExec = (script) => new Promise((resolve, reject) =>
+  exec(script, (err, stdout, stderr) => {
+    if (err) {
+      const errors = err.message.split('\n')
+      // ignore error if it's "key not found"
+      if (!(errors.length > 1 && errors[1] === ERR_KEY_NOT_FOUND)) {
+        return reject(err)
+      }
+    }
+    if (stderr) {
+      if (stderr !== ERR_KEY_NOT_FOUND) {
+        return reject(new KDBError(stderr))
+      } else {
+        return resolve() // no key found, return no value
+      }
+    }
+    const result = trimNewline(stdout)
+    if (!result) {
+      return resolve() // empty result, return no value
+    }
+    return resolve(result)
+  })
+)
+
+// escape strings by surrounding them with ""
+const escapeValues = (template, ...values) =>
+  template.reduce((acc, part, i) => {
+    let val = values[i - 1]
+    if (typeof val === 'string') val = `"${val}"`
+    return acc + val + part
+  })
+
+const ELEKTRA_VERSION_REGEX = /KDB_VERSION\:\ ([0-9]+\.[0-9]+\.[0-9]+)\n/
+
+// get Elektra version
+const version = () =>
+  safeExec(`kdb --version`)
+    .then(output => { // parse result
+      const matches = ELEKTRA_VERSION_REGEX.exec(output)
+      if (!matches) {
+        throw new Error('invalid version: ' + output)
+      }
+      return matches
+    })
+    .then(matches => matches.length > 1 && matches[1]) // select version from matches
+    .then(fullVersion => {
+      const splitVersions = fullVersion.split('.')
+      return {
+        version: fullVersion,
+        major: Number(splitVersions[0]),
+        minor: Number(splitVersions[1]),
+        micro: Number(splitVersions[2])
+      }
+    })
+
+// list available paths under a given `path`
+const ls = (path) =>
+  safeExec(escapeValues`kdb ls ${path}`)
+    .then(stdout => stdout && stdout.split('\n'))
+
+// get value from given `path`
+const get = (path) =>
+  safeExec(escapeValues`kdb get ${path}`)
+
+// set value at given `path`
+const set = (path, value) =>
+  safeExec(escapeValues`kdb set ${path} ${value}`)
+
+// remove value at given `path`
+const rm = (path) =>
+  safeExec(escapeValues`kdb rm ${path}`)
+
+// export javascript object from given `path`
+const _export = (path) =>
```

Comment:
@krit0n thanks a lot for the extensive guide and providing an environment to reproduce the error :ok_hand:

I tried it with the Dockerfile as I'm more familiar with docker than with vagrant, and it worked fine!

Now I could reproduce your issue, it does seem to be related to nodejs, because this works in your environment:

```
> kdb export "user/sw/elektra/web/#0/current/clusters" yajl
{

}
```

While these don't work (via `node` v6.9.4):

```
child_process.exec('kdb export "user/sw/elektra/web/#0/current/clusters" yajl', (err, stdout, stderr) => console.log(stderr)); '';
```

```
child_process.exec('/usr/local/bin/kdb export "user/sw/elektra/web/#0/current/clusters" yajl', (err, stdout, stderr) => console.log(stderr)); '';
```

Both of these commands result in:

```
Error (#75) occurred!
Description: Could not open file for writing.
Ingroup: plugin
Module: storage
At: /root/libelektra/src/plugins/yajl/yajl_gen.c:255
Reason: No such device or address
Mountpoint: user/sw/elektra/web/#0/current/clusters
Configfile: /dev/stdout
```

@markus2330 any idea what the issue could be here?